shared/cache.ts

```diff
@@ -0,0 +1,70 @@
+import { DateTime } from "luxon";
+
+type Cache<DataType> = {
+  value?: DataType;
+  updatedAt?: string;
+  timeToLive: number;
+  name?: string;
+};
+
+const defaultTimeToLive = 5 * 60;
+
+export const createCache = <DataType>(
+  getValue: () => Promise<DataType>,
+  defaultCache: Partial<Cache<DataType>> = {}
+) => {
+  let cache: Cache<DataType> = {
+    value: defaultCache.value ?? undefined,
+    updatedAt: defaultCache.updatedAt ?? undefined,
+    timeToLive: defaultCache.timeToLive ?? defaultTimeToLive,
+    name: defaultCache.name,
+  };
+
+  const getOrUpdateCache = async () => {
+    if (isCacheExpired(cache)) {
+      console.log(
+        `${cache.name ? `${cache.name} - ` : ""}Cache miss, updating.`
+      );
+
+      try {
+        const value = await getValue();
+
+        cache.value = value;
+        cache.updatedAt = DateTime.utc().toISO();
+
+        console.log(`${cache.name ? `${cache.name} - ` : ""}Cache updated.`);
+        return cache;
+      } catch (e: any) {
```

Comment:
Hehe `any`

Reply:
Hehe is there any (it you will) better way to do it?

Reply:
https://kentcdodds.com/blog/get-a-catch-block-error-message-with-typescript I suppose ;)

Reply:
Haha imo leave as is if ya want

Reply:
Might look at that in a future PR. Seems like it'd be good to have / know 🤩